### PR TITLE
Index oversize text for search.

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
PTAL @michaelkirk

### CODE REVIEW

File: SignalServiceKit/src/Storage/FullTextSearchFinder.swift [hunk 1]
```diff
@@ -170,6 +209,12 @@ public class FullTextSearchFinder: NSObject {
     }
 
     private class var dbExtensionConfig: YapDatabaseFullTextSearch {
+        SwiftAssertIsOnMainThread(#function)
+
+        if dbConnection == nil {
+            dbConnection = OWSPrimaryStorage.shared().newDatabaseConnection()
```
Comment (charlesmchen-signal) [inline SignalServiceKit/src/Storage/FullTextSearchFinder.swift]: You might not like this; I'm trying to ensure that the db connection gets initialized exactly once, on the main thread, and at a well-defined time.

Comment (michaelkirk-signal) [reply SignalServiceKit/src/Storage/FullTextSearchFinder.swift]: This is indeed a strange place for a side effect like this. Why not use the global read connection?

Comment (charlesmchen-signal) [reply SignalServiceKit/src/Storage/FullTextSearchFinder.swift]: Removed.  I thought you would prefer this way.

File: SignalServiceKit/src/Storage/FullTextSearchFinder.swift [hunk 2]
```diff
@@ -128,9 +128,48 @@ public class FullTextSearchFinder: NSObject {
...
+                return
+            }
+            guard let attachmentStream = attachment as? TSAttachmentStream else {
+                return
+            }
+            guard attachmentStream.isOversizeText() else {
+                return
+            }
+            Logger.verbose("attachmentStream: \(attachmentStream.contentType)")
+            Logger.flush()
```
[the remark this answers is not in the document]
Comment (charlesmchen-signal) [reply SignalServiceKit/src/Storage/FullTextSearchFinder.swift]: Done.

**Comment by charlesmchen-signal** [comment]
PTAL @michaelkirk

**Pull request merged**